next-docs/pages/templates/typescript/three.mdx

```diff
@@ -1 +1,256 @@
 # Three.js
+
+This template is for developers who use [Three.js](hhttps://threejs.org/).
+
+## Onchain code
+
+### `mud.config.ts`
+
+[This file](https://github.com/latticexyz/mud/blob/main/templates/threejs/packages/contracts/mud.config.ts) contains the table definition.
+
+```ts
+import { mudConfig } from "@latticexyz/world/register";
+
+export default mudConfig({
+  tables: {
+    Position: {
+      valueSchema: {
+        x: "int32",
+        y: "int32",
+        z: "int32",
+      },
+    },
+  },
+});
+```
+
+The table schema has one table, `Position`.
+This table has no specified key schema, so MUD uses the default, a single `bytes32` field.
+This allows us to tie table records to entities, which in MUD have a `bytes32` identifier.
+
+### `MoveSystem.sol`
+
+[This file](https://github.com/latticexyz/mud/blob/main/templates/threejs/packages/contracts/src/systems/MoveSystem.sol) is the system that is used to modify the data in MUD, which in this case is only `Position`.
+
+```solidity
+pragma solidity >=0.8.21;
+import { System } from "@latticexyz/world/src/System.sol";
+```
+
+The basic definition of what a `System` is.
+
+```solidity
+import { Position, PositionData } from "../codegen/index.sol";
+```
+
+Get the definitions related to the `Position` table, which are automatically generated from `mud.config.ts`.
+
+```solidity
+function distance(PositionData memory a, PositionData memory b) pure returns (int32) {
+  int32 deltaX = a.x > b.x ? a.x - b.x : b.x - a.x;
+  int32 deltaY = a.y > b.y ? a.y - b.y : b.y - a.y;
+  int32 deltaZ = a.z > b.z ? a.z - b.z : b.z - a.z;
+
+  return deltaX + deltaY + deltaZ;
+}
+```
+
+A utility function to calculate [Manhattan Distance](https://en.wikipedia.org/wiki/Taxicab_geometry).
+This is used to ensure that a player does not move more than one unit at a time (in any direction).
+
+```solidity
+contract MoveSystem is System {
+  function move(int32 x, int32 y, int32 z) public {
+```
+
+This is the only function that `MoveSystem` exposes.
+It lets a user move in game universe.
+
+```solidity
+    bytes32 entityId = bytes32(uint256(uint160((_msgSender()))));
+```
+
+`_msgSender()` lets us know who called the world which received the call and then relayed it to the system.
```

Comment:
```suggestion
`_msgSender()` lets us know who called the `World` which received the call and then relayed it to the system.
```
let's generally write is like this to make it clear that this is a contract (could be confusing otherwise with limited context)